### ISSUE TITLE
router 3.1.0 with ES5: TypeError: Cannot read property 'filter' of undefined

### ISSUE THREAD

**Issue by albertosantini** [angular/angular#12266]
**I'm submitting a ...**  (check one with "x")
```
[X ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
The app throws the following exception:

```
VM1853 core.umd.js:3263 TypeError: Cannot read property 'filter' of undefined
    at RouterPreloader.setUpPreloading (https://unpkg.com/@angular/router@3.1.0/bundles/router.umd.js:3556:51)
    at https://unpkg.com/@angular/router@3.1.0/bundles/router.umd.js:3791:23
    at https://unpkg.com/@angular/core@2.1.0/bundles/core.umd.js:6825:60
    at Array.forEach (native)
    at ApplicationRef_._loadComponent (https://unpkg.com/@angular/core@2.1.0/bundles/core.umd.js:6825:23)
    at ApplicationRef_.bootstrap (https://unpkg.com/@angular/core@2.1.0/bundles/core.umd.js:6811:18)
    at https://unpkg.com/@angular/core@2.1.0/bundles/core.umd.js:6713:93
    at Array.forEach (native)
    at PlatformRef_._moduleDoBootstrap (https://unpkg.com/@angular/core@2.1.0/bundles/core.umd.js:6713:46)
    at https://unpkg.com/@angular/core@2.1.0/bundles/core.umd.js:6681:31
```

**Expected behavior**
View1 should be rendered displaying `View View1 loaded with the router`.

**Minimal reproduction of the problem with instructions**

With angular2 2.1.0, router 3.1.0, rxjs 5.0.0-beta.12, it doesn't work:
http://plnkr.co/edit/1IFDIOQD8z405Zm3oasr?p=preview

I tried also withr rxjs 5.0.0-rc.1 and it doesn't work.

With angular2 2.1.0, router 3.0.2, rxjs 5.0.0-beta.12, it works correctly:
http://plnkr.co/edit/IblBTr799g3DJSlclckQ?p=preview

Similar to https://github.com/angular/angular/issues/11300.

In this case rxjs_operator_filter is undefined.

https://unpkg.com/@angular/router@3.1.0/bundles/router.umd.js:3556:51
var navigations = rxjs_operator_filter.filter.call(this.router.events, function (e) { return e instanceof NavigationEnd; });

**What is the motivation / use case for changing the behavior?**

Upgrade from angular 2.0.2 (router 3.0.2, rxjs 5.0.0-beta.12) to angular 2.1.0 (router 3.1.0)

**Please tell us about your environment:**

* **Angular version:** 2.1.0 and router 3.1.0

* **Browser:** [all]

 
* **Language:** [ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by robwormald** [comment]
Appears like this comes from the Observable.prototype.filter operator not being excluded in the rollup build here: https://github.com/angular/angular/blob/master/modules/%40angular/router/rollup.config.js#L30-L39

This would be an easy PR if anybody wants to attack it :)

cc @vsavkin

**Comment by hstarorg** [comment]
Update ``router.umd.js`` can fix the bug.

```javascript
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('rxjs/Subject'), require('rxjs/observable/from'), require('rxjs/observable/of'), require('rxjs/operator/concatMap'), require('rxjs/operator/every'), require('rxjs/operator/map'), require('rxjs/operator/mergeAll'), require('rxjs/operator/mergeMap'), require('rxjs/operator/reduce'), require('rxjs/Observable'), require('rxjs/operator/catch'), require('rxjs/operator/concatAll'), require('rxjs/operator/first'), require('rxjs/util/EmptyError'), require('rxjs/observable/fromPromise'), require('rxjs/operator/last'), require('rxjs/BehaviorSubject'), require('rxjs/operator/filter')) :
        typeof define === 'function' && define.amd ? define(['exports', '@angular/common', '@angular/core', 'rxjs/Subject', 'rxjs/observable/from', 'rxjs/observable/of', 'rxjs/operator/concatMap', 'rxjs/operator/every', 'rxjs/operator/map', 'rxjs/operator/mergeAll', 'rxjs/operator/mergeMap', 'rxjs/operator/reduce', 'rxjs/Observable', 'rxjs/operator/catch', 'rxjs/operator/concatAll', 'rxjs/operator/first', 'rxjs/util/EmptyError', 'rxjs/observable/fromPromise', 'rxjs/operator/last', 'rxjs/BehaviorSubject', 'rxjs/operator/filter'], factory) :
            (factory((global.ng = global.ng || {}, global.ng.router = global.ng.router || {}),
                global.ng.common, global.ng.core,
                global.Rx,
                global.Rx.Observable,
                global.Rx.Observable,
                global.Rx.Observable.prototype, // this one
                global.Rx.Observable.prototype,
                global.Rx.Observable.prototype,
                global.Rx.Observable.prototype,
                global.Rx.Observable.prototype,
                global.Rx.Observable.prototype,
                global.Rx,
                global.Rx.Observable.prototype,
                global.Rx.Observable.prototype, global.Rx.Observable.prototype, global.Rx, global.Rx.Observable,
                global.Rx.Observable.prototype, global.Rx,
                global.Rx.Observable.prototype // this one
            ));
} (this, function (exports, _angular_common, _angular_core, rxjs_Subject, rxjs_observable_from, rxjs_observable_of, rxjs_operator_concatMap, rxjs_operator_every, rxjs_operator_map, rxjs_operator_mergeAll, rxjs_operator_mergeMap, rxjs_operator_reduce, rxjs_Observable, rxjs_operator_catch, rxjs_operator_concatAll, rxjs_operator_first, rxjs_util_EmptyError, rxjs_observable_fromPromise, l, rxjs_BehaviorSubject, rxjs_operator_filter) {
    'use strict';
```

**Comment by afnpires** [comment]
Is there a estimated date for the fix release?

**Comment by hstarorg** [comment]
@afnpires the bug is fixed but don't know when it release, so you can use my way to temporary resolve it.

**Comment by afnpires** [comment]
@hstarorg I'm sorry but your workaround doesn't seem to work.
![router umd js](https://cloud.githubusercontent.com/assets/5302351/19457223/3f82c3e0-94bd-11e6-90f7-68a59b112d84.jpg)

**Comment by hstarorg** [comment]
@afnpires You can try this file that I have processed.
[router.umd.txt](https://github.com/angular/angular/files/535213/router.umd.txt)

**Comment by afnpires** [comment]
@hstarorg didn't work. Same error.

**Comment by mh-dev** [comment]
Your problem is related to https://github.com/angular/angular/issues/9359
Something with the bundled RX.js does not work.

**Comment by atiertant** [comment]
@afnpires @hstarorg works for me with https://github.com/angular/angular/issues/12266#issuecomment-253733661

**Comment by hstarorg** [comment]
@atiertant :)

**Comment by CorvusCorrax** [comment]
New package @angular/router 3.1.1 have still the bug with rxjs concatMap

`EXCEPTION: rxjs_operator_concatMap is undefined`

Fix : https://github.com/angular/angular/pull/12430

**Comment by afnpires** [comment]
@hstarorg finnaly I understand why it didn't work for me.

Instead of specifying the rxjs in system.config.js map (1) I was using paths (2), so I could avoid loading each rxjs operation individually.


(1)
```
map: {
      // ...
      'rxjs':                      'npm:rxjs'
    },
```

(2)
```
 paths: {
      'npm:': 'node_modules/',
      'rxjs/*': 'node_modules/rxjs/bundles/Rx.min.js'
    },
```

**Comment by hstarorg** [comment]
But the 3.1.1 has a newer problem, that ``concatMap`` can not working. We also should modify the umd code.

**Comment by ericmartinezr** [comment]
@hstarorg there's already a PR to fix that https://github.com/angular/angular/pull/12430

**Comment by hstarorg** [comment]
@ericmartinezr But have no release version.

**Comment by albertosantini** [comment]
That `concatMap` PR has not merged in 3.1.2 yet.
Stucked to 3.0.2 due to this kind of errors.

**Comment by hstarorg** [comment]
@albertosantini Bad idea.
